src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBaseSourceConfig.java

```diff
@@ -137,13 +160,21 @@ protected SalesforceBaseSourceConfig(String referenceName,
     this.duration = duration;
     this.offset = offset;
     this.operation = operation;
+    this.initialRetryDuration = initialRetryDuration;
+    this.maxRetryDuration = maxRetryDuration;
+    this.retryOnBackendError = retryOnBackendError;
+    this.maxRetryCount = maxRetryCount;
   }
 
 
   public Map<ChronoUnit, Integer> getDuration() {
     return extractRangeValue(SalesforceSourceConstants.PROPERTY_DURATION, duration);
   }
 
+  public Boolean isRetryRequired() {
+    return retryOnBackendError;
```

Comment:
This would be null in case of older pipelines. Did we verify it?

Reply:
done

Reply:
this change has already been done in  SalesforceBulkRecordReader
 isRetryRequired = Boolean.valueOf(conf.get(SalesforceSourceConstants.CONFIG_RETRY_REQUIRED, String.valueOf(true)));

Reply:
return retryOnBackendError == null || retryOnBackendError need to change like this?

Reply:
My intent was that we should handle it somewhere at the top level or in this class. 

I leave it up to you to decide.

Reply:
Done the change

Reply:
> My intent was that we should handle it somewhere at the top level or in this class.
> 
> I leave it up to you to decide.

ya changed it in config as well